lib/index.js

```diff
@@ -40,20 +40,25 @@ if (typeof process.env.NODE_PG_FORCE_NATIVE !== 'undefined') {
   module.exports = new PG(Client)
 
   // lazy require native module...the native module may not have installed
-  module.exports.__defineGetter__('native', function () {
-    delete module.exports.native
-    var native = null
-    try {
-      native = new PG(require('./native'))
-    } catch (err) {
-      if (err.code !== 'MODULE_NOT_FOUND') {
-        throw err
+  var native
+  Object.defineProperty(module.exports, 'native', {
+    enumerable: false,
+    get () {
+      if (native) {
+        return native
       }
-      /* eslint-disable no-console */
-      console.error(err.message)
-      /* eslint-enable no-console */
+
+      try {
```

Comment:
To avoid redoing this (and printing the error) every time it’s accessed, the original approach is good:

```js
Object.defineProperty(module.exports, 'native', {
  value: native,
})
```

(`value` replaces `get`/`set`)

Reply:
@charmander Not sure I follow. Wouldn't that immediately load `./native`, rather than lazily load it?

As far as I can tell, the original approach only works because it overwrites `module.exports.native` after the first invocation of the getter [1].

[1] https://github.com/brianc/node-postgres/blob/8f56b8c2fd97119f23215af543eec90bbcee959b/lib/index.js#L56

Reply:
A second `Object.defineProperty` inside the getter, I mean. `{ value }` will get merged with the existing descriptor to produce `{ configurable: true, value: native }`.

Reply:
Oh, you already wrote a comment about that: https://github.com/brianc/node-postgres/pull/1992/files#r336177268. (`delete` should work when the property is configurable, but like you said, it’s not necessary.)

Reply:
@charmander Updated. Let me know if that's what you had in mind.